hpecp/k8s_worker.py

```diff
@@ -297,31 +297,48 @@ def item_not_exists():
             except polling.TimeoutException:
                 return False
 
-    def set_storage(self, worker_id, persistent_disks, ephemeral_disks):
-        """Set Storage
+    def set_storage(self, worker_id, ephemeral_disks, persistent_disks=[]):
+        """Set storage for a k8s worker.
+
+        Parameters
+        ----------
+        worker_id : str
+            The k8s worker ID, format - '/api/v2/worker/k8shost/[0-9]+'
+        ephemeral_disks : list
+            List of ephemeral disks. Mandatory parameter.
+        persistent_disks : list, optional
+            List of persistent disks, by default []
 
-        :param k8sworker_id: the worker ID
-        :param persistent_disks: a comma separated list of zero or more
-        persistent disks, e.g. "/dev/nvme2n1"
-        :param ephemeral_disks: a comma separated list of zero or more
-        ephemeral_disks disks, e.g. "/dev/nvme1n1"
+        Raises
+        ------
+        APIItemNotFoundException
+        APIItemConflictException
+        APIException
+        AssertionError
         """
+        # Make sure that the worker exists
+        self.get(worker_id)
 
-        # TODO: id is valid
-        # TODO: assert disks are arrays
+        assert isinstance(
+            ephemeral_disks, list
+        ), "'ephemeral_disks' must be provided and and must be a list"
+        assert isinstance(
+            persistent_disks, list
+        ), "'persistent_disks' must be a list"
 
+        # Prepare the payload
         data = {
             "op_spec": {
-                "persistent_disks": persistent_disks,
                 "ephemeral_disks": ephemeral_disks,
+                "persistent_disks": persistent_disks,
             },
             "op": "storage",
         }
 
-        self.client._request(
+        # Make the request
+        _response = self.client._request(
```

Comment:
are you intending to use the _response for something?

Reply:
Nope, removed that in a newer commit. Converted this PR to a draft, adding unit-tests, will open it up for review.

Github actions do work :)

Reply:
great job!